server/src/AMQPConnectionInfo.cc

```diff
@@ -32,6 +34,7 @@ using namespace mlpl;
 struct AMQPConnectionInfo::Impl {
 	Impl()
 	: m_URL(),
+	  m_URLBuf(NULL),
```

Comment:
amqp_parse_url(url, connectionInfo) do the following things:

* Replace some characters in url with "\0" to split the url string to url elements
* Store pointer to each url elements into connectionInfo. It doesn't copy the strings, refer them directly.